src/BalanceTx.purs

```diff
@@ -437,36 +452,36 @@ balanceTx unattachedTx@(UnattachedUnbalancedTx { unbalancedTx: t }) = do
     logTx "Unbalanced Collaterised Tx " allUtxos unbalancedCollTx
     -- Prebalance collaterised tx without fees:
     ubcTx <- except $
-      prebalanceCollateral zero allUtxos ownAddr unbalancedCollTx
+      prebalanceCollateral zero allUtxos ownAddr utxoMinVal unbalancedCollTx
     -- Prebalance collaterised tx with fees:
     let unattachedTx' = unattachedTx # _transaction' .~ ubcTx
     _ /\ fees <- ExceptT $ evalExUnitsAndMinFee unattachedTx'
     ubcTx' <- except $
-      prebalanceCollateral (fees + feeBuffer) allUtxos ownAddr ubcTx
+      prebalanceCollateral (fees + feeBuffer) allUtxos ownAddr utxoMinVal ubcTx
     -- Loop to balance non-Ada assets
     nonAdaBalancedCollTx <- ExceptT $ loop allUtxos ownAddr [] $ unattachedTx' #
       _transaction' .~ ubcTx'
     -- Return excess Ada change to wallet:
     unsignedTx <- ExceptT $
       returnAdaChangeAndFinalizeFees ownAddr allUtxos nonAdaBalancedCollTx <#>
         lmap ReturnAdaChangeError'
-    let
-      unattachedTx'' = unsignedTx ^. _transaction'
-        /\ unsignedTx ^. _redeemersTxIns
-      -- Sort inputs at the very end so it behaves as a Set.
-      sortedUnsignedTx = fst unattachedTx'' # _body <<< _inputs %~ Array.sort
-    -- Logs final balanced tx and returns it
-    logTx "Post-balancing Tx " allUtxos sortedUnsignedTx
-    except $ Right (unattachedTx'' # _1 .~ sortedUnsignedTx)
+    -- Sort inputs at the very end so it behaves as a Set:
+    let sortedUnsignedTx = unsignedTx # _body' <<< _inputs %~ Array.sort
```

Comment:
Is it possible to use Set instead?

Reply:
I've been thinking about this for a long time, I will create an issue for this and can open discussion.